transport/src/main/java/com/strategyobject/substrateclient/transport/ws/BackoffReconnectionPolicy.java

```diff
@@ -0,0 +1,126 @@
+package com.strategyobject.substrateclient.transport.ws;
+
+import com.google.common.base.Preconditions;
+import lombok.*;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Represents a backoff retry policy
+ */
+@RequiredArgsConstructor(access = AccessLevel.PRIVATE)
+@Getter
+@Slf4j
+public class BackoffReconnectionPolicy implements ReconnectionPolicy {
```

Comment:
`Backoff` doesn't tell me anything about the algorithm being used. How about `ExponentialReconnectionPolicy`?

Reply:
The policy allows to use a constant time backoff also that's why I called it like this. I think "exponential backoff" is more common name so will rename to `ExponentialBackoffPolicy` or `ExponentialBackoffReconnectionPolicy`.